tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterBase.java

```diff
@@ -147,47 +127,40 @@ public void testAdapterRejectsBadInboundMessage(final TestContext context) {
      * 
      * @param context The Vert.x test context.
      */
-    @Test
+    @Test(timeout = 4000)
     public void testUploadMessageFailsForTenantDisabledAdapter(final TestContext context) {
-        final Async completionTracker = context.async();
 
         final String tenantId = helper.getRandomTenantId();
         final String deviceId = helper.getRandomDeviceId(tenantId);
-        final String username = IntegrationTestSupport.getUsername(deviceId, tenantId);
-        final String password = "secret";
-
-        final TenantObject tenant = TenantObject.from(tenantId, true);
-        tenant.addAdapterConfiguration(TenantObject.newAdapterConfig(Constants.PROTOCOL_ADAPTER_TYPE_AMQP, false));
 
-        final Async addTenantTracker = context.async();
+        setupProtocolAdapter(tenantId, deviceId, true, context);
+        final Message msg = ProtonHelper.message("some payload");
+        sender.send(msg).setHandler(context.asyncAssertFailure(t -> {
 
-        helper.registry
-                .addDeviceForTenant(tenant, deviceId, password)
-                .setHandler(context.asyncAssertSuccess(ok -> addTenantTracker.complete()));
-
-        addTenantTracker.await();
-
-        final HonoClient client = prepareAmqpAdapterClient(username, password);
-        client.connect(new ProtonClientOptions()).compose(ok -> {
-            clientsToShutdown.add(client);
-            return getOrCreateSender(tenantId, client);
-        }).compose(sender -> {
-            final Message msg = ProtonHelper.message("some payload");
-            return sender.send(msg);
-        }).setHandler(context.asyncAssertFailure(t -> {
             context.assertTrue(ClientErrorException.class.isInstance(t));
 
             // HonoClient maps AmqpError.UNAUTHORIZED_ACCESS to HttpURLConnection.HTTP_BAD_REQUEST
             // -> so we test the message
 
-            final String expectedErrorMsg = String
-                    .format("This adapter is not enabled for tenant [tenantId: %s].", tenantId);
+            final String expectedErrorMsg = String.format("This adapter is not enabled for tenant [tenantId: %s].",
+                    tenantId);
 
             context.assertEquals(expectedErrorMsg, ((ClientErrorException) t).getMessage());
+        }));
+    }
 
-            completionTracker.complete();
+    /**
+     * Verifies that the AMQP Adapter will fail to authenticate a device whose username does not match the expected pattern
+     * {@code [<authId>@<tenantId>]}.
+     * 
+     * @param context The Vert.x test context.
+     */
+    @Test(timeout = 1000)
+    public void testAuthenticationWithInvalidUsernamePattern(final TestContext context) {
+        connectToAdapter("invalidaUsername", DEVICE_PASSWORD, true)
+        .setHandler(context.asyncAssertFailure(t -> {
+            context.assertTrue(ServiceInvocationException.class.isInstance(t));
```

Comment:
why not check for ClientErrorException and status code 401?

Reply:
yes, make sense